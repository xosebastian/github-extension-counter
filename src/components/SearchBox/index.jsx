import * as React from 'react';
import PropTypes from 'prop-types';
import { useState } from 'react';
import Grid from '@mui/material/Grid';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';

export default function SearchBox({ onSearch, onOrder }) {
  const [order, setOrder] = useState('asc');

  const onChangeOrder = (value) => {
    setOrder(value);
    onOrder(value);
  };

  return (
    <Box sx={{ flexGrow: 1, mt: 2 }}>
      <Grid container spacing={1}>
        <Grid item xs={10}>
          <TextField
            fullWidth
            size="small"
            autoComplete="off"
            id="owner-basic"
            label="Search..."
            variant="outlined"
            onChange={(e) => onSearch(e.target.value)}
          />
        </Grid>
        <Grid item xs={2}>
          <Select
            size="small"
            fullWidth
            value={order}
            label="OrderBy"
            onChange={(e) => onChangeOrder(e.target.value)}
          >
            <MenuItem value="asc">Asc </MenuItem>
            <MenuItem value="desc">Desc </MenuItem>
          </Select>
        </Grid>
      </Grid>
    </Box>
  );
}

SearchBox.propTypes = {
  onSearch: PropTypes.func.isRequired,
  onOrder: PropTypes.func.isRequired,
};
