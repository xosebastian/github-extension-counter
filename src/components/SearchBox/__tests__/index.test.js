/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import SearchBox from "../index";

const mockOnSearch = jest.fn();
const mockOnOrder = jest.fn();

const getRenderedComponent = () =>
  render(<SearchBox onSearch={mockOnSearch} onOrder={mockOnOrder} />);

describe("<SearchBox/>", () => {
  it("Must render the component", () => {
    const { baseElement } = getRenderedComponent();
    expect(baseElement).toMatchSnapshot();
  });

  it("Expect to not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error");
    getRenderedComponent();
    expect(spy).not.toHaveBeenCalled();
  });
});
