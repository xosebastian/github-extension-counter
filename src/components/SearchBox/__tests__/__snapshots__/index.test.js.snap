// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox/> Must render the component 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-be40y1"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="owner-basic"
              id="owner-basic-label"
            >
              Search...
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="owner-basic"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Search...
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-sizeSmall  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              Asc 
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="asc"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  OrderBy
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
